admin/services/data-access/sql.role-connection.builder.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const roles = require('../../lib/consts/roles')
+const sqlConfig = require('../../config/sql.config')
+const config = require('../../config')
+const R = require('ramda')
+
+const builder = {
+  /**
+   * @description builds a sql connection config object
+   * @param {string} roleName - the role of the user to create a connection config for
+   * @param {boolean} readonly - set to true for a readonly connection, typically to a sql azure replica
+   * @returns {object} a config object that works with mssql
+   */
+  build: function build (roleName, readonly = false) {
+    let cfg
+    switch (roleName) {
+      case roles.teacher:
+        cfg = sqlConfig
```

Comment:
This is assigning a reference to sqlConfig, and there is a line doing mutation later on.

Suggest we freeze sqlConfig once it is read, and return a cloned copy here.
`cfg = R.clone(sqlConfig)`

Reply:
great idea, a lot safer as it could be overlooked in further down the line